### ISSUE TITLE
Add an Album to Favorites goes wrong.

### ISSUE THREAD

**Issue by BeActivated** [CDrummond/lms-material#586]
Hi,

If there are several albums in the LMS collection with the same Title but by a different Artist it adds always the same Album to the favorites, but that is not the right one (except for the one case).
Example:
I add 'Anthology' from 'The Clean' to the favorites, but In the favorites appears 'Anthology' from 'The Babys' 

Thanks for your work.

**Comment by CDrummond** [comment]
Sorry, but that is an LMS limitation. When you favourite and album LMS saves it as a DB search on the album name.

Any fix would require changes in LMS.

**Comment by CDrummond** [comment]
I'm going to re-open this. I see that LMS has been updated to also save the album_artist when favouriting albums  - and for albums from online-services it stores their 'external id'. Material should, therefore, reflect this behaviour - but I _think_ this is LMS 8.3 specific, so any change should be for that version only.

**Comment by michaelherger** [comment]
Came here to report this issue - to find @BeActivated's report. Thanks for getting back on this, @CDrummond. I figured out I could do something like this:

```
diff --git a/MaterialSkin/HTML/material/html/js/utils.js b/MaterialSkin/HTML/material/html/js/utils.js
index fe4a5daf..3776d4f4 100644
--- a/MaterialSkin/HTML/material/html/js/utils.js
+++ b/MaterialSkin/HTML/material/html/js/utils.js
@@ -743,7 +743,7 @@ function updateItemFavorites(item) {
             item.favUrl="db:contributor.name="+encodeURIComponent(favTitle);
             item.favIcon=changeImageSizing(item.image);
         } else if (item.id.startsWith("album_id:")) {
-            item.favUrl="db:album.title="+encodeURIComponent(favTitle);
+            item.favUrl="db:album.title="+encodeURIComponent(favTitle)+"&contributor.name="+encodeURIComponent(item.artists[0]);
             item.favIcon=changeImageSizing(item.image);
         } else if (item.id.startsWith("year:")) {
             item.favUrl="db:year.id="+encodeURIComponent(favTitle);
```

But that likely would break compatibility with older LMS versions.

**Comment by CDrummond** [comment]
@michaelherger Yeah, that is where I would make any change. Not sure if "item.artists" is always valid then. e.g. when browsing My Music -> Album Artists - > Artist -> Album I don't think the album items here have artists (as not required). Anyhow, that's my problem to solve :)

**Comment by michaelherger** [comment]
> Anyhow, that's my problem to solve :)

I like that attitude 😁!

**Comment by CDrummond** [comment]
Fixed for 2.9.5
